ui/src/features/annotator/hooks/use-active-tool.hook.ts

```diff
@@ -0,0 +1,16 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+import { useState } from 'react';
+
+import { ToolType } from '../../../components/tool-selection-bar/tools/interface';
+
+export const useActiveTool = () => {
+    const [activeTool, setActiveTool] = useState<ToolType>('selection');
+
+    const handleSelectTool = (tool: ToolType) => {
+        setActiveTool(tool);
+    };
+
+    return { activeTool, selectTool: handleSelectTool };
```

Comment:
np:

```suggestion
    const [activeTool, setActiveTool] = useState<ToolType>('selection');

    return { activeTool, selectTool: setActiveTool };
```

(perhaps rename `selectTool` to `setActiveTool`? I don't see a huge benefit atm of abstracting this)

Reply:
Updated